Fix windows build error

Fix windows build error : https://github.com/qualcomm/eld/actions/runs/25064591249/job/73428067425

D:\a\eld\eld\nightly\llvm-project\llvm\tools\eld\include\eld\Readers\Relocation.h(101,8): error C2039: 'string': is not a member of 'std' [D:\a\eld\eld\nightly\obj\tools\eld\lib\Readers\ELDReaders.vcxproj]
  (compiling source file '../../../../../llvm-project/llvm/tools/eld/lib/Readers/Relocation.cpp')
      D:\a\eld\eld\nightly\llvm-project\llvm\include\llvm\ADT\STLExtras.h(2707,11):
      see declaration of 'std'

--- a/include/eld/Readers/Relocation.h
+++ b/include/eld/Readers/Relocation.h
@@ -12,10 +12,10 @@
 //===----------------------------------------------------------------------===//
 #ifndef ELD_READERS_RELOCATION_H
 #define ELD_READERS_RELOCATION_H
+
 #include "eld/Config/Config.h"
 #include "eld/Fragment/FragmentRef.h"
-#include "llvm/ADT/DenseMap.h"
-#include "llvm/Support/DataTypes.h"
+#include <string>
 #include <unordered_map>
 
 namespace eld {
@@ -29,7 +29,6 @@ class ELFSection;
 class ELFSegment;
 
 class Relocation {
-  static llvm::DenseMap<const Relocation *, FragmentRef *> m_RelocFragMap;
 
 public:
   typedef uint64_t Address; // FIXME: use SizeTrait<T>::Address instead
@@ -115,18 +114,9 @@ public:
 
   void setTargetData(DWord pTargetData);
 
-  void modifyRelocationFragmentRef(FragmentRef *fragRef) {
-    if (!fragRef)
-      return;
-    m_RelocFragMap[this] = fragRef;
-  }
+  void modifyRelocationFragmentRef(FragmentRef *fragRef);
 
-  FragmentRef *targetFragRef() const {
-    auto it = m_RelocFragMap.find(this);
-    if (it == m_RelocFragMap.end())
-      return nullptr;
-    return it->second;
-  }
+  FragmentRef *targetFragRef() const;
 
   static std::string getFragmentPath(ResolveInfo *symInfo, Fragment *frag,
                                      const GeneralOptions &options);

--- a/lib/Readers/Relocation.cpp
+++ b/lib/Readers/Relocation.cpp
@@ -15,6 +15,8 @@
 #include "eld/SymbolResolver/LDSymbol.h"
 #include "eld/SymbolResolver/ResolveInfo.h"
 #include "eld/Target/Relocator.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/Support/DataTypes.h"
 #include "llvm/Support/ManagedStatic.h"
 
 namespace eld {
@@ -23,7 +25,7 @@ class DiagnosticEngine;
 
 using namespace eld;
 
-llvm::DenseMap<const Relocation *, FragmentRef *> Relocation::m_RelocFragMap;
+static llvm::DenseMap<const Relocation *, FragmentRef *> m_RelocFragMap;
 
 //===----------------------------------------------------------------------===//
 // Relocation Factory Methods
@@ -315,6 +317,19 @@ void Relocation::setSymInfo(ResolveInfo *pSym) { m_pSymInfo = pSym; }
 
 void Relocation::setTargetData(DWord pTargetData) {
   m_TargetData = pTargetData;
+}
+
+void Relocation::modifyRelocationFragmentRef(FragmentRef *fragRef) {
+  if (!fragRef)
+    return;
+  m_RelocFragMap[this] = fragRef;
+}
+
+FragmentRef *Relocation::targetFragRef() const {
+  auto it = m_RelocFragMap.find(this);
+  if (it == m_RelocFragMap.end())
+    return nullptr;
+  return it->second;
 }
 
 Relocation::Size Relocation::size(Relocator &pRelocator) const {

--- a/test/Common/standalone/PluginAPI/ELDExpected.test
+++ b/test/Common/standalone/PluginAPI/ELDExpected.test
@@ -3,7 +3,7 @@
 # This test checks the functioning of eld::Expected.
 #END_COMMENT
 #START_TEST
-RUN: %llvmobjroot/bin/tests%config/ELDExpectedUsage%var | %filecheck %s
+RUN: %llvmobjroot/bin/tests/%config/ELDExpectedUsage%var | %filecheck %s
 #END_TEST
 #CHECK: e1.has_value(): 1
 #CHECK: e1.value(): 11

--- a/test/Common/standalone/linkerscript/SysrootExpansion/SysrootExpansion.test
+++ b/test/Common/standalone/linkerscript/SysrootExpansion/SysrootExpansion.test
@@ -1,3 +1,4 @@
+#UNSUPPORTED: windows
 #---SysrootExpansion.test---------------- Linker Script ----------------#
 #BEGIN_COMMENT
 # Validate sysroot expansion for "=" and "$SYSROOT" markers in INPUT/GROUP

--- a/test/Common/standalone/sysRoot/VersionAndDynamicScripts/VersionAndDynamicScriptAbsolutePath.test
+++ b/test/Common/standalone/sysRoot/VersionAndDynamicScripts/VersionAndDynamicScriptAbsolutePath.test
@@ -1,3 +1,4 @@
+#UNSUPPORTED: windows
 #---VersionAndDynamicScripts.test---- sysRoot ----------------#
 
 #BEGIN_COMMENT

--- a/test/lit.cfg
+++ b/test/lit.cfg
@@ -574,7 +574,7 @@ def find_config_dir(base_path):
     # Try common configuration names
     for config_name in ['Release', 'Debug', 'RelWithDebInfo', 'MinSizeRel']:
         if (Path(base_path) / config_name).exists():
-            return os.sep + config_name
+            return config_name
     # No config subdirectory found (single-config generator)
     return ""
 

